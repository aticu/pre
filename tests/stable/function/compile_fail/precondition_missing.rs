use pre::pre;

#[pre("must be bar")]
#[pre("must be baz")]
fn foo() {}

#[pre]
fn main() {
    #[assert_pre("must be bar", reason = "is bar")]
    foo()
}
