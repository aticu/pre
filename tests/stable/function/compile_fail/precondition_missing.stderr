error[E0063]: missing field `_custom_must_20be_20baz` in initializer of `foo`
  --> $DIR/precondition_missing.rs:10:5
   |
10 |     foo()
   |     ^^^ missing `_custom_must_20be_20baz`
